src/audio/kpb.c

```diff
@@ -171,45 +179,56 @@ static int kpb_prepare(struct comp_dev *dev)
 
 	cd->no_of_clients = 0;
 
-	/* allocate history_buffer/s */
-#if KPB_NO_OF_HISTORY_BUFFERS == 2
+	/* allocate history buffer/s */
+	while (hb_size > 0 && i < ARRAY_SIZE(hb_mcp)) {
+		ptr = rballoc(RZONE_RUNTIME, hb_mcp[i], ca_size);
+
+		if (ptr) {
+			history_buffer->start_addr = ptr;
+			history_buffer->end_addr = ptr + ca_size;
+			history_buffer->w_ptr = ptr;
+			history_buffer->r_ptr = ptr;
+			history_buffer->state = KPB_BUFFER_FREE;
+			hb_size = hb_size - ca_size;
+			history_buffer->next = &cd->history_buffer;
```

Comment:
is this intentional? Shouldnt the next point to NULL?

Reply:
@ranj063, where do you see a NULL assignment? No, neither buffer should point t o NULL...there is one big circular buffer so it must have its predecessor and successor.